src/main/java/mat/client/measure/ManageMeasurePresenter.java

```diff
@@ -564,16 +591,48 @@ public void onSuccess(List<? extends HasListBox> result) {
                 detailDisplay.setScoringChoices(result);
             }
         });
-
         detailDisplay.getMeasureScoringListBox().addChangeHandler(event -> setPatientBasedIndicatorBasedOnScoringChoice((detailDisplay)));
-
-        detailDisplay.getMeasureNameTextBox().addValueChangeHandler(event -> setIsPageDirty(true));
-        detailDisplay.getECQMAbbreviatedTitleTextBox().addValueChangeHandler(event -> setIsPageDirty(true));
-        detailDisplay.getCQLLibraryNameTextBox().addValueChangeHandler(event -> setIsPageDirty(true));
         detailDisplay.getMeasureScoringListBox().addValueChangeHandler(event -> setIsPageDirty(true));
         detailDisplay.getPatientBasedListBox().addValueChangeHandler(event -> setIsPageDirty(true));
     }
 
+    private void addHandlerToGenerateCmsId() {
+        detailDisplay.getFhirModel().addValueChangeHandler(clickEvent -> {
+            if (clickEvent.getValue()) {
+                detailDisplay.getGenerateCmsIdCheckbox().setText("Automatically generate a CMS ID and matching Library Name upon save");
+                detailDisplay.getGenerateCmsIdCheckbox().setTitle("Click to generate a CMS ID and matching Library Name upon save");
+            }
+        });
+        detailDisplay.getQdmModel().addValueChangeHandler(clickEvent -> {
+            if (clickEvent.getValue()) {
+                detailDisplay.getGenerateCmsIdCheckbox().setText("Automatically generate a eCQM ID and matching Library Name upon save");
+                detailDisplay.getGenerateCmsIdCheckbox().setTitle("Automatically generate a eCQM ID and matching Library Name upon save");
```

Comment:
Does the text and title need to be set twice in two separate places?

Reply:
'AbstractNewMeasureView' sets the initial Text and Title, This handler will change text based on Measure model radio button selection.

Reply:
Gotcha, thanks for clarifying!